### ISSUE TITLE
Document and automate libjsonnet.js regeneration

### ISSUE THREAD

**Issue by mbrukman** [yosuke-furukawa/node-jsonnet#2]
It appears that [`libjsonnet.js`](https://github.com/yosuke-furukawa/node-jsonnet/blob/master/lib/libjsonnet.js) is either fully auto-generated from Emscripten from Jsonnet C++ source or possibly with some manual modifications of auto-generated code.

It would be nice to document how this is done so that it's easy to upgrade this module as Jsonnet changes, and also to automate it as much as possible so that it's easy for anyone to update it in the future.

**Comment by loretoparisi** [comment]
@mbrukman The auto generation is done as explained in `jsonnet` javascript implementation page:

[Javascript Implementation](http://jsonnet.org/implementation/javascript.html)
-  You need `emscripten` installed on your system first: http://kripken.github.io/emscripten-site/docs/getting_started/downloads.html
- Then you take the [C++ implementation](http://jsonnet.org/implementation/cpp.html) from Github: https://github.com/google/jsonnet
- Build

**Comment by sparkprime** [comment]
That's right.  Long term, it would be interesting to explore using gopherjs to build a js implementation from the native Go port, which is still a work-in-progress.  The reason being that Go has a garbage collector so Jsonnet does not need its own.  The go implementation will also use native stacks instead of a manually-managed Jsonnet stack within the interpreter.  However it'll be interesting to see how well the go unbounded stacks transfer to Javascript's hard limited stacks.

**Comment by Rush** [comment]
Any progress on this one?

**Comment by buremba** [comment]
Any plans for the js implementation from the native Go port? @sparkprime
